File: Projects/TransportathonHackathon.API/TransportathonHackathon.Application/Features/Employees/Queries/GetByCompanyId/GetByCompanyIdEmployeeResponse.cs
using TransportathonHackathon.Domain.Entities;

namespace TransportathonHackathon.Application.Features.Employees.Queries.GetByCompanyId
{
    public class GetByCompanyIdEmployeeResponse
    {
        public Guid AppUserId { get; set; }
        public Guid CompanyId { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Age { get; set; }
        public bool IsOnTransitNow { get; set; }

        public virtual Carrier? Carrier { get; set; }
        public virtual Driver? Driver { get; set; }
    }
}

File: Projects/TransportathonHackathon.Clients/angular-ui/src/app/models/response-models/employees/getByCompanyIdEmployeeResponse.ts
import { Carrier } from '../../domain-models/carrier';
import { Driver } from '../../domain-models/driver';

export interface GetByCompanyIdEmployeeResponse {
  appUserId: string;
  companyId: string;
  firstName: string;
  lastName: string;
  age: number;
  isOnTransitNow: boolean;
  carrier: Carrier | null;
  driver: Driver | null;
}
